modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -928,5 +944,136 @@ PolygonConcentricCircleMeshGeneratorBase::generate()
   if (flat_side_up)
     MeshTools::Modification::rotate(*mesh0, 180.0 / (Real)_num_sides, 0.0, 0.0);
   mesh0->set_isnt_prepared();
+
+  if (_has_rings && getParam<bool>("replace_inner_ring_with_delaunay_mesh"))
+  {
+    if (getParam<bool>("quad_center_elements"))
+      paramWarning("quad_center_elements",
+                   "ignored because the inner ring will be replaced by a Delaunay mesh with "
+                   "'replace_inner_ring_with_delaunay_mesh=true'");
+
+    // remove elements of the inner ring and create an inner-ring external boundary side set
+    std::set<Elem *> deleteable_elems;
+    for (auto & elem : mesh0->element_ptr_range())
+      if (elem->subdomain_id() == block_ids_new[0] || elem->subdomain_id() == block_ids_new[1])
```

Comment:
the inner ring includes block_ids_new[1] only when `_ring_inner_boundary_layer_params.intervals.front() + _ring_intervals.front() + _ring_outer_boundary_layer_params.intervals.front()>1`,

Reply:
The test seems running okay. Is this condition the case for the test?

Reply:
The test does run well. If you have `_ring_intervals[0]>1`, this `if` will work fine. Otherwise, this will cause issues.

Reply:
Actually hold on, I remembered I choose my original logic because rings may have the same block ids specified by users, even the background region could be the same. I think the original logic is safe typically. Do you have a better solution for this?

Reply:
You are right, the block ids might not be unique.

If you want to use the old approach, you need to use `ring_radii_corr` instead of `_ring_radii`.

OR, if the replacement is used, modify the first or first/second elements of `block_ids_new` to max(block_ids_new)+1/+2 before `assignBlockIdsNames()`

Reply:
I like the later approach. It can be easily done.

Reply:
The last approach will have implication on the block names. It is a lot easier to just check the radius.